<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get started • pareg</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get started">
<meta property="og:description" content="pareg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pareg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/pareg.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/pathway_similarities.html">Pathway similarities</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/cbg-ethz/pareg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="pareg_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get started</h1>
                        <h4 data-toc-skip class="author">Kim Philipp Jablonski</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cbg-ethz/pareg/blob/HEAD/vignettes/pareg.Rmd" class="external-link"><code>vignettes/pareg.Rmd</code></a></small>
      <div class="hidden name"><code>pareg.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>This vignette is an introduction to the usage of <code>pareg</code>. It estimates pathway enrichment scores by regressing differential expression p-values of all genes considered in an experiment on their membership to a set of biological pathways. These scores are computed using a regularized generalized linear model with LASSO and network regularization terms. The network regularization term is based on a pathway similarity matrix (e.g., defined by Jaccard similarity) and thus classifies this method as a modular enrichment analysis tool <span class="citation">(Huang, Sherman, and Lempicki 2009)</span>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"pareg"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-required-packages">Load required packages<a class="anchor" aria-label="anchor" href="#load-required-packages"></a>
</h2>
<p>We start our analysis by loading the <code>pareg</code> package and other required libraries.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">enrichplot</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cbg-ethz/pareg" class="external-link">pareg</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introductory-example">Introductory example<a class="anchor" aria-label="anchor" href="#introductory-example"></a>
</h2>
<div class="section level3">
<h3 id="generate-pathway-database">Generate pathway database<a class="anchor" aria-label="anchor" href="#generate-pathway-database"></a>
</h3>
<p>For the sake of this introductory example, we generate a synthetic pathway database with a pronounced clustering of pathways.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">group_num</span> <span class="op">&lt;-</span> <span class="fl">2</span>
<span class="va">pathways_from_group</span> <span class="op">&lt;-</span> <span class="fl">10</span>

<span class="va">gene_groups</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">group_num</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">group_idx</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu">glue</span><span class="fu">::</span><span class="fu"><a href="https://glue.tidyverse.org/reference/glue.html" class="external-link">glue</a></span><span class="op">(</span><span class="st">"g{group_idx}_gene_{seq_len(15)}"</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>
<span class="va">genes_bg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"bg_gene_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">50</span><span class="op">)</span><span class="op">)</span>

<span class="va">df_terms</span> <span class="op">&lt;-</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/imap.html" class="external-link">imap_dfr</a></span><span class="op">(</span>
  <span class="va">gene_groups</span>,
  <span class="kw">function</span><span class="op">(</span><span class="va">current_gene_list</span>, <span class="va">gene_list_idx</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_dfr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">pathways_from_group</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">pathway_idx</span><span class="op">)</span> <span class="op">{</span>
      <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
        term <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"g"</span>, <span class="va">gene_list_idx</span>, <span class="st">"_term_"</span>, <span class="va">pathway_idx</span><span class="op">)</span>,
        gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
          <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">current_gene_list</span>, <span class="fl">10</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,
          <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">genes_bg</span>, <span class="fl">10</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
        <span class="op">)</span>
      <span class="op">)</span>
    <span class="op">}</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">)</span>

<span class="va">df_terms</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/sample_n.html" class="external-link">sample_n</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##        term       gene</span>
<span class="co">## 1 g1_term_9 g1_gene_12</span>
<span class="co">## 2 g1_term_5  g1_gene_7</span>
<span class="co">## 3 g2_term_2  g2_gene_2</span>
<span class="co">## 4 g1_term_3 bg_gene_47</span>
<span class="co">## 5 g1_term_8  g1_gene_1</span></code></pre>
</div>
<div class="section level3">
<h3 id="term-similarities">Term similarities<a class="anchor" aria-label="anchor" href="#term-similarities"></a>
</h3>
<p>Before starting the actual enrichment estimation, we compute pairwise pathway similarities with <code>pareg</code>’s helper function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mat_similarities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_term_similarities.html">compute_term_similarities</a></span><span class="op">(</span>
  <span class="va">df_terms</span>,
  similarity_function <span class="op">=</span> <span class="va">jaccard</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">mat_similarities</span>, xlab <span class="op">=</span> <span class="st">"Term similarity"</span><span class="op">)</span></code></pre></div>
<p><img src="pareg_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>We can see a clear clustering of pathways.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span>
  <span class="va">mat_similarities</span>,
  name <span class="op">=</span> <span class="st">"Similarity"</span>,
  col <span class="op">=</span> <span class="fu">circlize</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/circlize/man/colorRamp2.html" class="external-link">colorRamp2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"black"</span><span class="op">)</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="pareg_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="create-synthetic-study">Create synthetic study<a class="anchor" aria-label="anchor" href="#create-synthetic-study"></a>
</h3>
<p>We then select a subset of pathways to be activated. In a performance evaluation, these would be considered to be true positives.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">active_terms</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/similarity_sample.html">similarity_sample</a></span><span class="op">(</span><span class="va">mat_similarities</span>, <span class="fl">5</span><span class="op">)</span>
<span class="va">active_terms</span></code></pre></div>
<pre><code><span class="co">## [1] "g2_term_6" "g2_term_3" "g2_term_3" "g2_term_2" "g2_term_8"</span></code></pre>
<p>The genes contained in the union of active pathways are considered to be differentially expressed.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_genes</span> <span class="op">&lt;-</span> <span class="va">df_terms</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">term</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">active_terms</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span>

<span class="va">other_genes</span> <span class="op">&lt;-</span> <span class="va">df_terms</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">de_genes</span><span class="op">)</span></code></pre></div>
<p>The p-values of genes considered to be differentially expressed are sampled from a Beta distribution centered at <span class="math inline">\(0\)</span>. The p-values for all other genes are drawn from a Uniform distribution.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">de_genes</span>, <span class="va">other_genes</span><span class="op">)</span>,
  pvalue <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Beta.html" class="external-link">rbeta</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">de_genes</span><span class="op">)</span>, <span class="fl">0.1</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/stats/Beta.html" class="external-link">rbeta</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">other_genes</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>,
  in_study <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">TRUE</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">de_genes</span><span class="op">)</span><span class="op">)</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">other_genes</span><span class="op">)</span><span class="op">)</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span>
  <span class="va">df_study</span><span class="op">$</span><span class="va">pvalue</span> <span class="op">&lt;=</span> <span class="fl">0.05</span>,
  <span class="va">df_study</span><span class="op">$</span><span class="va">in_study</span>, dnn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sig. p-value"</span>, <span class="st">"in study"</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">##             in study</span>
<span class="co">## sig. p-value FALSE TRUE</span>
<span class="co">##        FALSE    34   17</span>
<span class="co">##        TRUE      1   28</span></code></pre>
</div>
<div class="section level3">
<h3 id="enrichment-analysis">Enrichment analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h3>
<p>Finally, we compute pathway enrichment scores.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pareg.html">pareg</a></span><span class="op">(</span>
  <span class="va">df_study</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gene</span>, <span class="va">pvalue</span><span class="op">)</span>,
  <span class="va">df_terms</span>,
  network_param <span class="op">=</span> <span class="fl">1</span>, term_network <span class="op">=</span> <span class="va">mat_similarities</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Loaded Tensorflow version 2.3.0</span></code></pre>
<p>The results can be exported to a dataframe for further processing…</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">enrichment</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">term</th>
<th align="right">enrichment</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">g2_term_6</td>
<td align="right">-0.6760687</td>
</tr>
<tr class="even">
<td align="left">g2_term_3</td>
<td align="right">-0.6005415</td>
</tr>
<tr class="odd">
<td align="left">g2_term_2</td>
<td align="right">-0.5818557</td>
</tr>
<tr class="even">
<td align="left">g2_term_4</td>
<td align="right">-0.4233026</td>
</tr>
<tr class="odd">
<td align="left">g2_term_8</td>
<td align="right">-0.4123425</td>
</tr>
<tr class="even">
<td align="left">g1_term_2</td>
<td align="right">0.3978593</td>
</tr>
</tbody>
</table>
<p>…and also visualized in a pathway network view.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit</span>, min_similarity <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></code></pre></div>
<p><img src="pareg_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>To provide a wider range of visualization options, the result can be transformed into an object which is understood by the functions of the <a href="https://github.com/YuLab-SMU/enrichplot" class="external-link">enrichplot</a> package.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/as_enrichplot_object.html">as_enrichplot_object</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/dotplot.html" class="external-link">dotplot</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_colour_continuous</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Enrichment Score"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'colour' is already present. Adding another scale for 'colour',</span>
<span class="co">## which will replace the existing scale.</span></code></pre>
<p><img src="pareg_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/enrichplot/man/treeplot.html" class="external-link">treeplot</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_colour_continuous</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"Enrichment Score"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Scale for 'colour' is already present. Adding another scale for 'colour',</span>
<span class="co">## which will replace the existing scale.</span></code></pre>
<p><img src="pareg_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.2.0 (2022-04-22)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur/Monterey 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] grid      stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] pareg_0.99.5          tfprobability_0.15.0  tensorflow_2.8.0     </span>
<span class="co">##  [4] enrichplot_1.16.1     ComplexHeatmap_2.12.0 forcats_0.5.1        </span>
<span class="co">##  [7] stringr_1.4.0         dplyr_1.0.9           purrr_0.3.4          </span>
<span class="co">## [10] readr_2.1.2           tidyr_1.2.0           tibble_3.1.7         </span>
<span class="co">## [13] tidyverse_1.3.1       ggraph_2.0.5          ggplot2_3.3.6        </span>
<span class="co">## [16] BiocStyle_2.24.0     </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2             reticulate_1.25-9000   tidyselect_1.1.2      </span>
<span class="co">##   [4] RSQLite_2.2.14         AnnotationDbi_1.58.0   BiocParallel_1.30.2   </span>
<span class="co">##   [7] devtools_2.4.3         scatterpie_0.1.7       munsell_0.5.0         </span>
<span class="co">##  [10] codetools_0.2-18       ragg_1.2.2             future_1.25.0         </span>
<span class="co">##  [13] withr_2.5.0            keras_2.8.0            colorspace_2.0-3      </span>
<span class="co">##  [16] GOSemSim_2.22.0        Biobase_2.56.0         highr_0.9             </span>
<span class="co">##  [19] logger_0.2.2           knitr_1.39             rstudioapi_0.13       </span>
<span class="co">##  [22] stats4_4.2.0           DOSE_3.22.0            listenv_0.8.0         </span>
<span class="co">##  [25] labeling_0.4.2         GenomeInfoDbData_1.2.8 polyclip_1.10-0       </span>
<span class="co">##  [28] bit64_4.0.5            farver_2.1.0           rprojroot_2.0.3       </span>
<span class="co">##  [31] parallelly_1.31.1      vctrs_0.4.1            treeio_1.20.0         </span>
<span class="co">##  [34] generics_0.1.2         xfun_0.31              R6_2.5.1              </span>
<span class="co">##  [37] doParallel_1.0.17      GenomeInfoDb_1.32.2    clue_0.3-60           </span>
<span class="co">##  [40] graphlayouts_0.8.0     bitops_1.0-7           cachem_1.0.6          </span>
<span class="co">##  [43] fgsea_1.22.0           gridGraphics_0.5-1     assertthat_0.2.1      </span>
<span class="co">##  [46] scales_1.2.0           gtable_0.3.0           globals_0.15.0        </span>
<span class="co">##  [49] processx_3.5.3         tidygraph_1.2.1        rlang_1.0.2           </span>
<span class="co">##  [52] zeallot_0.1.0          systemfonts_1.0.4      GlobalOptions_0.1.2   </span>
<span class="co">##  [55] splines_4.2.0          lazyeval_0.2.2         broom_0.8.0           </span>
<span class="co">##  [58] BiocManager_1.30.18    yaml_2.3.5             reshape2_1.4.4        </span>
<span class="co">##  [61] modelr_0.1.8           backports_1.4.1        qvalue_2.28.0         </span>
<span class="co">##  [64] usethis_2.1.5          tools_4.2.0            bookdown_0.26         </span>
<span class="co">##  [67] ggplotify_0.1.0        ellipsis_0.3.2         jquerylib_0.1.4       </span>
<span class="co">##  [70] RColorBrewer_1.1-3     proxy_0.4-26           BiocGenerics_0.42.0   </span>
<span class="co">##  [73] sessioninfo_1.2.2      Rcpp_1.0.8.3           plyr_1.8.7            </span>
<span class="co">##  [76] progress_1.2.2         base64enc_0.1-3        zlibbioc_1.42.0       </span>
<span class="co">##  [79] RCurl_1.98-1.6         ps_1.7.0               prettyunits_1.1.1     </span>
<span class="co">##  [82] GetoptLong_1.0.5       viridis_0.6.2          S4Vectors_0.34.0      </span>
<span class="co">##  [85] haven_2.5.0            ggrepel_0.9.1          cluster_2.1.3         </span>
<span class="co">##  [88] here_1.0.1             fs_1.5.2               magrittr_2.0.3        </span>
<span class="co">##  [91] data.table_1.14.2      DO.db_2.9              circlize_0.4.15       </span>
<span class="co">##  [94] reprex_2.0.1           whisker_0.4            ggnewscale_0.4.7      </span>
<span class="co">##  [97] matrixStats_0.62.0     pkgload_1.2.4          hms_1.1.1             </span>
<span class="co">## [100] patchwork_1.1.1        evaluate_0.15          readxl_1.4.0          </span>
<span class="co">## [103] IRanges_2.30.0         gridExtra_2.3          shape_1.4.6           </span>
<span class="co">## [106] tfruns_1.5.0           testthat_3.1.4         compiler_4.2.0        </span>
<span class="co">## [109] crayon_1.5.1           shadowtext_0.1.2       htmltools_0.5.2       </span>
<span class="co">## [112] ggfun_0.0.6            tzdb_0.3.0             aplot_0.1.4           </span>
<span class="co">## [115] lubridate_1.8.0        DBI_1.1.2              tweenr_1.0.2          </span>
<span class="co">## [118] dbplyr_2.1.1           MASS_7.3-56            Matrix_1.4-1          </span>
<span class="co">## [121] brio_1.1.3             cli_3.3.0              parallel_4.2.0        </span>
<span class="co">## [124] igraph_1.3.1           pkgconfig_2.0.3        pkgdown_2.0.3         </span>
<span class="co">## [127] xml2_1.3.3             foreach_1.5.2          ggtree_3.4.0          </span>
<span class="co">## [130] bslib_0.3.1            rngtools_1.5.2         XVector_0.36.0        </span>
<span class="co">## [133] doFuture_0.12.2        rvest_1.0.2            doRNG_1.8.2           </span>
<span class="co">## [136] yulab.utils_0.0.4      callr_3.7.0            digest_0.6.29         </span>
<span class="co">## [139] Biostrings_2.64.0      rmarkdown_2.14         cellranger_1.1.0      </span>
<span class="co">## [142] fastmatch_1.1-3        tidytree_0.3.9         nloptr_2.0.2          </span>
<span class="co">## [145] rjson_0.2.21           lifecycle_1.0.1        nlme_3.1-157          </span>
<span class="co">## [148] jsonlite_1.8.0         desc_1.4.1             viridisLite_0.4.0     </span>
<span class="co">## [151] fansi_1.0.3            pillar_1.7.0           lattice_0.20-45       </span>
<span class="co">## [154] KEGGREST_1.36.0        fastmap_1.1.0          httr_1.4.3            </span>
<span class="co">## [157] pkgbuild_1.3.1         GO.db_3.15.0           remotes_2.4.2         </span>
<span class="co">## [160] glue_1.6.2             png_0.1-7              iterators_1.0.14      </span>
<span class="co">## [163] bit_4.0.4              ggforce_0.3.3          stringi_1.7.6         </span>
<span class="co">## [166] sass_0.4.1             blob_1.2.3             textshaping_0.3.6     </span>
<span class="co">## [169] memoise_2.0.1          ape_5.6-2</span></code></pre>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references">
<div id="ref-huang2009bioinformatics">
<p>Huang, Da Wei, Brad T Sherman, and Richard A Lempicki. 2009. “Bioinformatics Enrichment Tools: Paths Toward the Comprehensive Functional Analysis of Large Gene Lists.” <em>Nucleic Acids Research</em> 37 (1): 1–13.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kim Philipp Jablonski.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
